devito/yask/transformer.py

```diff
@@ -0,0 +1,178 @@
+        for i in tree:
+            if not i.dim.is_Sub:
+                continue
+
+            ydim = nfac.new_domain_index(i.dim.parent.name)
+
+            try:
+                # Are the extremes statically known ?
+                lower = nfac.new_const_number_node(int(i.dim.symbolic_ofs_lower))
+                upper = nfac.new_const_number_node(int(i.dim.symbolic_ofs_upper))
+
+                # Yes! Can go on adding the conditions
+                expr = nfac.new_add_node(nfac.new_first_domain_index(ydim), lower)
+                for _, v in conditions:
+                    v.append(nfac.new_not_less_than_node(ydim, expr))
+                expr = nfac.new_add_node(nfac.new_last_domain_index(ydim), upper)
+                for _, v in conditions:
+                    v.append(nfac.new_not_greater_than_node(ydim, expr))
+            except TypeError:
+                # Nope, is at least the Iteration extent statically known? If not,
+                # than a further TypeError exception will be thrown by ``i.extent()``
+                if i.is_Sequential:
+                    extent = i.extent()
+
+                    if i.direction is Backward:
+                        _range = reversed(range(extent))
+                        node = nfac.new_first_domain_index(ydim)
+                    elif i.direction is Forward:
+                        _range = range(-extent + 1, 1)
+                        node = nfac.new_last_domain_index(ydim)
+                    else:
+                        assert False
+
+                    unwinded = []
+                    for e, v in conditions:
+                        for r in _range:
+                            expr = nfac.new_add_node(node, nfac.new_const_number_node(r))
+                            condition = nfac.new_equals_node(ydim, expr)
+                            unwinded.append((e, v + [nfac.new_equals_node(ydim, expr)]))
+                    conditions = unwinded
+                else:
+                    # YASK does not support parallel Iterations over sub-dimensions
+                    # with statically unknown extents
+                    raise NotImplementedError
+
+        # Build the YASK equations as well as all necessary grids
+        for k, v in conditions:
+            yask_expr = handle(k, yc_soln, mapper)
+
+            if yask_expr is not None:
+                processed.append(yask_expr)
+
+                # Is there a sub-domain to attach ?
+                if v:
+                    condition = v.pop(0)
+                    for i in v:
+                        condition = nfac.new_and_node(condition, i)
+                    yask_expr.set_cond(condition)
+
+    # Add flow dependences to the offloaded equations
+    # TODO: This can be improved by spotting supergroups ?
+    for to, frm in zip(processed, processed[1:]):
+        yc_soln.add_flow_dependency(frm, to)
+
+    return mapper
+
+
+def handle(expr, yc_soln, mapper):
```

Comment:
note for reviewers: this function is nothing new, it's just the formerly called ``sympy2yask``